crates/router/src/connector/mifinity/transformers.rs

```diff
@@ -124,109 +278,147 @@ impl<F, T>
             types::PaymentsResponseData,
         >,
     ) -> Result<Self, Self::Error> {
+        let payload = item.response.payload.first();
```

Comment:
Throw error here if payload is not present, instead of using `.map` on top of it